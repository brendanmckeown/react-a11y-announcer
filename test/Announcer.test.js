import Adapter from 'enzyme-adapter-react-16';
import Announcer from '../src/Announcer';
import React from 'react';
import { configure, shallow, mount } from 'enzyme';

configure({ adapter: new Adapter() });

test('Component is created', () => {
  const wrapper = mount(<Announcer text="This is an announcement!" />);
  let tree = wrapper.html();
  expect(tree).toMatchSnapshot();
});

test('Should have a default politeness prop of `polite`', () => {
  const wrapper = shallow(<Announcer text="" />);
  const politeness = wrapper.instance().props.politeness;
  expect(politeness).toBe('polite');
});

test('Text node updates when state changes', () => {
  // Since we really can't test the live announcement,
  // we can at least ensure the text node gets updated.
  const wrapper = shallow(<Announcer text="" />);
  const testAnnouncementStr = 'Here\'s a new announcement!';
  expect(wrapper.text()).toEqual('');
  wrapper.setProps({ text: testAnnouncementStr });
  expect(wrapper.text()).toEqual(testAnnouncementStr);
});
