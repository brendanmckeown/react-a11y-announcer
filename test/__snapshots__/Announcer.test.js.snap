// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is created 1`] = `"<div aria-atomic=\\"true\\" aria-live=\\"polite\\" style=\\"position: absolute; visibility: visible; overflow: hidden; display: block; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; clip: rect(0px, 0px, 0px, 0px); clip-path: polygon(0px 0px, 0px 0px, 0px 0px, 0px 0px); white-space: nowrap;\\" class=\\"\\"></div>"`;
